app/com/arpnetworking/metrics/portal/alerts/impl/FileAlertRepository.java

```diff
@@ -0,0 +1,403 @@
+                .filter(containsPredicate)
+                .count();
+
+        return new DefaultQueryResult<>(alerts, total);
+    }
+
+    @Override
+    public long getAlertCount(final Organization organization) {
+        assertIsOpen();
+        return _alerts.size();
+    }
+
+    /* Unsupported mutation operations */
+
+    @Override
+    public int deleteAlert(final UUID identifier, final Organization organization) {
+        // Since we expect to use this repository just as every other AlertRepository,
+        // we should enforce the open-before-use invariant rather than immediately
+        // throwing on mutations.
+        assertIsOpen();
+        throw new UnsupportedOperationException("FilesystemAlertRepository is read-only");
+    }
+
+    @Override
+    public void addOrUpdateAlert(final Alert alert, final Organization organization) {
+        assertIsOpen();
+        throw new UnsupportedOperationException("FilesystemAlertRepository is read-only");
+    }
+
+    private ImmutableMap<UUID, Alert> loadAlerts() {
+        final AlertGroup group;
+        try (Reader reader = Files.newBufferedReader(_path)) {
+            group = _objectMapper.readValue(
+                    reader,
+                    AlertGroup.class);
+        } catch (final IOException e) {
+            throw new RuntimeException("Could not load alerts", e);
+        }
+        final ImmutableMap.Builder<UUID, Alert> mapBuilder = ImmutableMap.builder();
+        for (final SerializedAlert fsAlert : group.getAlerts()) {
+            final UUID uuid = fsAlert.getUUID().orElseGet(() -> computeUUID(fsAlert));
+
+            // TODO(cbriones):
+            // Unlike an ordinary query, the alert start and end times are dependent
+            // on both the current time and query itself, so it's likely that this
+            // query field needs to be expressed differently.
+            final models.internal.MetricsQuery query = new DefaultMetricsQuery.Builder()
+                    .setQuery(fsAlert.getQuery())
+                    .setStart(ZonedDateTime.now())
+                    .setEnd(ZonedDateTime.now())
+                    .build();
+            final Alert alert =
+                    new DefaultAlert.Builder()
+                            .setId(uuid)
+                            .setName(fsAlert.getName())
+                            .setDescription(fsAlert.getDescription())
+                            .setEnabled(fsAlert.isEnabled())
+                            .setOrganization(_organization)
+                            .setQuery(query)
+                            .setAdditionalMetadata(fsAlert.getAdditionalMetadata())
+                            .build();
+            mapBuilder.put(uuid, alert);
+        }
+        return mapBuilder.build();
+    }
+
+    private UUID computeUUID(final SerializedAlert alert) {
+        // TODO(cbriones): UUID v5 rather than v3.
+        //
+        // v3 UUID identifiers use MD5 while v5 uses SHA-1.
```

Comment:
This seems like it will become much harder to fix once we push this code for the first time, since the UUIDs will change for all the alerts that don't specify UUIDs. True? Or am I missing something?

Reply:
I was thinking that if anything ever required us to change how we created the UUID anyway we could "freeze" it into the serialized form. _but_ I will still see if I can do anything about this in this diff.

The main hiccup is that JDK8 doesn't support UUID v5 so I'll either need to copy-pasta this in or pull in another dep.

Reply:
"freeze it" == "take all the UUIDless alert definitions that are imported into MPortal, and give them explicit v3 UUIDs"?

Reply:
Yes, freeze would mean to take the existing UUIDs they already have and assign them to the definitions so that they don't get computed at read time. Then we would be free to change how they are computed for newer definitions. I think we would do all of this on the config side though outside of MPortal's runtime.

Reply:
Hrrm... if it were _just us_ who needed to do that, I'd be on board, but if it's _all_ users of MPortal alerts, I feel like a backwards-incompatible change is likely to get shot down. It seems pretty easy to pull in a library that would give us UUIDv5, though -- am I mistaken?